### DISCUSSION TITLE
illegal instruction

### DISCUSSION THREAD

**Discussion by ZephyrLi-pro** [ptitSeb/box64#978] [Q&A]
Hi, petitSeb. I encountered an illegal instruction error while running an application after executing 'Jr a3' once. I would like to know how to troubleshoot this issue.

**Comment by ptitSeb** [comment]
You should use a TRACE enable build of box64 and use `BOX64_DYNAREC_DUMP=2 BOX64_TRACE=0-1` BOX64_TRACE_FILE=dump.txt` to create a dump of the generated code (this will be more readable you have libZyDis.so on your system).
That way, you can try to understand what is happening.

Note that if the illegal instruction is randomly hapenning, it might be a multi-threading issue. In that case, I don't have any easy way to debug this kind of things :(

**Reply by ZephyrLi-pro** [reply]
OK, thanks
